source/Nevermore/Advanced/ReadTransaction.cs

```diff
@@ -540,20 +577,23 @@ protected async Task<TResult[]> ReadResultsAsync<TResult>(PreparedCommand prepar
         {
             var mapping = configuration.DocumentMaps.Resolve(typeof(TDocument));
 
+            if (mapping.IdColumn is null)
```

Comment:
We could mark IdColumn as not nullable  if owe passed it to DocumentMap via the constructor. I can imagine this will break gazillion things in the server? https://github.com/OctopusDeploy/Nevermore/blob/e21a19848c32e44a0038fbd80b3190c3d495d14f/source/Nevermore/Mapping/DocumentMap.cs#L258